internal/common/conversion/flatten_expand.go

```diff
@@ -28,6 +30,31 @@ func FlattenTags(tags []admin.ResourceTag) []map[string]string {
 	return ret
 }
 
+// TODO: remove once new SDK is available with cipher config changes
+func FlattenTagsPreview(tags []adminPreview.ResourceTag) []map[string]string {
+	ret := make([]map[string]string, len(tags))
```

Comment:
you might want to explore the use of CopyModel that I did with @EspenAlbert , for instance you can convert from adminPreview.ResourceTag to admin.ResourceTag and back, and in general between any two structs with the same fields: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/61eda15846a6623b9b4906bed1e7d7c1793504d8/internal/common/conversion/model_generation.go#L9C6-L9C15
we'll be discussing it in the next tech meeting

Reply:
resolving as removed usage of preview SDK